ouroboros-network/src/Ouroboros/Network/DeltaQ.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+module Ouroboros.Network.DeltaQ (
+    -- * GSV type
+    GSV(..),
+    Distribution,
+
+    -- * DeltaQ calculations
+    SizeInBytes,
+    DeltaQ,
+    gsvLeadingEdgeArrive,
+    gsvTrailingEdgeDepart,
+    gsvTrailingEdgeArrive,
+    deltaqQ99thPercentile,
+    deltaqProbabilityMassBeforeDeadline,
+
+    -- * Bi-directional GSV
+    PeerGSV(..),
+    gsvRequestResponseDuration,
+  ) where
+
+import           Data.Semigroup ((<>))
+import           Control.Monad.Class.MonadTimer (TimeMeasure, Duration)
+
+
+--
+-- Sketch of API for GSV and DeltaQ
+--
+
+data Distribution n = Distribution n
+-- This is a totally bogus representation. It's just a PoC.
+-- This says that there's a single value that it takes with probability 1.
+
+instance Num n => Semigroup (Distribution n) where
+  (<>) = convolveDistribution
+
+convolveDistribution :: Num n => Distribution n -> Distribution n -> Distribution n
+convolveDistribution (Distribution d) (Distribution d') = Distribution (d+d')
+-- Again, totally bogus.
+
+shiftDistribution :: Num n => n -> Distribution n -> Distribution n
+shiftDistribution n (Distribution d) = Distribution (n+d)
+-- Again, totally bogus.
+
+data GSV t = GSV (Duration t)                -- G as seconds
+                 (Duration t)                -- S as seconds / octet
```

Comment:
Comment says seconds / octet, but the type is s plain Duration. Is it the propagation time for one byte?

Reply:
it should be a function from size -> Duration t; representing it as _service time_ per octet is a reasonable starting approximation

Reply:
Yes, this is the duration for a single octet. Yes this is the simple model where we assume S is directly proportional to the size, hence time for a single octet.

Reply:
I have made it a function of size.